pkg/storaged/utils.js

```diff
@@ -682,16 +790,34 @@ export function get_active_usage(client, path, top_action, child_action) {
             return actions;
         }
 
+        function enter_unmount(block, location, is_top) {
+            for (const u of usage) {
+                if (u.usage == 'mounted' && u.location == location) {
+                    if (is_top) {
+                        u.actions = get_actions(_("unmount"));
+                        u.set_noauto = false;
+                    }
+                    return;
+                }
+            }
+            usage.push({
+                level,
+                block,
+                usage: 'mounted',
+                location,
+                set_noauto: !is_top,
+                actions: is_top ? get_actions(_("unmount")) : [_("unmount")],
+                blocking: false
+            });
+        }
+
         if (fsys && fsys.MountPoints.length > 0) {
             fsys.MountPoints.forEach(mp => {
-                usage.push({
-                    level,
-                    usage: 'mounted',
-                    block,
-                    location: decode_filename(mp),
-                    actions: get_actions(_("unmount")),
-                    blocking: false,
-                });
+                const mpd = decode_filename(mp);
+                const children = find_children_for_mount_point(client, mpd, null);
+                for (const c in children)
+                    enter_unmount(children[c], c, false);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19388-20230928-071616-0731117d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)